torch/csrc/jit/interned_strings.h

```diff
@@ -3,146 +3,252 @@
 #include <stdint.h>
 #include <string>
 #include <unordered_map>
+#include <algorithm>
 
 #include "torch/csrc/jit/generated/aten_interned_strings.h"
 
 namespace torch { namespace jit {
 
-// JIT symbols are synthetic operators that occur only in the JIT IR
+// Every symbol is classified in a namespace, specifying what kind of symbol it
+// is.  Unsigned char to ensure widening to unique_t (also an unsigned type)
+enum class SymbolNamespace : unsigned char {
+  onnx  = 'o',
+  prim  = 'p',
+  aten  = 't',
+  // NB: ONNX and ATen attributes all live in a global namespace, as
+  // their interpretation depends on the operator name (which is namespaced)
+  attr  = 'a',
+  // TODO: eliminate me
+  scope = 's'
+};
+
+namespace detail {
+
+  // TODO: programatically generate this
+  inline std::string valid_namespaces_str() {
+    return "'onnx', 'prim', 'aten', 'attr', 'scope'";
+  }
+
+}
+
+// Primitive symbols are synthetic operators that occur only in the IR
 // and don't have corresponding implementations in ATen.
 //
 // TODO: We need documentation for all of these symbols.
+//
+// TODO: Consider moving the synthetic onnx operators to their own
+// namespace.
```

Comment:
Do all prim symbols live in the "scope" namespace?

Reply:
I don't know what you mean by this. prim symbols live in the "prim" namespace.